### PR TITLE
WOWL: add tests to generic dropdown component

### PR DESCRIPTION
**Pull request by brboi**
Taskid: 2366896

### CODE REVIEW

File: addons/wowl/static/tests/components/dropdown_tests.ts [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+
+let env: OdooEnv;
+let parent: owl.Component;
+
+QUnit.module("Dropdown", {
+  async beforeEach() {
+    env = await makeTestEnv();
+  },
+  async afterEach() {
+    parent && parent.unmount();
```
Comment (aab-odoo) [inline addons/wowl/static/tests/components/dropdown_tests.ts]: I think that my linter doesn't like this

Comment (aab-odoo) [inline addons/wowl/static/tests/components/dropdown_tests.ts]:
```suggestion
    if (parent) {
        parent.unmount();
    }
```

File: addons/wowl/static/tests/components/dropdown_tests.ts [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+  await click(parent.el!, "button.o_dropdown_toggler");
+  assert.containsOnce(parent.el!, "ul.o_dropdown_menu");
+  await click(parent.el!, "button.o_dropdown_toggler");
+  assert.containsNone(parent.el!, "ul.o_dropdown_menu");
+});
+
+QUnit.test("initial open state can be true", async (assert) => {
+  class Parent extends owl.Component {
+    static components = { Dropdown };
+    static template = owl.tags.xml`<Dropdown open="true"/>`;
```
Comment (aab-odoo) [inline addons/wowl/static/tests/components/dropdown_tests.ts]: I thought we decided to rename this props

File: addons/wowl/static/tests/components/dropdown_tests.ts [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+          </t>
+        </Dropdown>
+      </div>
+    `;
+  }
+  parent = await mount(Parent, { env });
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "div.outside");
```
Comment (aab-odoo) [inline addons/wowl/static/tests/components/dropdown_tests.ts:191]: as a sanity check, assert above that 3 dropdowns are open

File: addons/wowl/static/tests/components/dropdown_tests.ts [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+      assert.deepEqual(ev.detail.payload, { answer: 42 });
+    }
+  }
+  parent = await mount(Parent, { env });
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+  await click(parent.el!, "button.o_dropdown_toggler:last-child");
+
+  // As two listeners are defined in the template,
+  // clicking once the item would execute the handler twice.
+  assert.expect(2);
```
Comment (aab-odoo) [inline addons/wowl/static/tests/components/dropdown_tests.ts]: typically, we define this as first line of the test

**Review by aab-odoo — COMMENTED** [review]
A few minor remarks, but the suite already looks solid :+1:

File: addons/wowl/static/tests/components/dropdown_tests.ts (hunk 2 above)
Comment (brboi) [reply addons/wowl/static/tests/components/dropdown_tests.ts]: That was not clear, I thought we decided this props was well documented :-D
What do you suggest: "startOpen" ?

File: addons/wowl/static/tests/components/dropdown_tests.ts (hunk 1 above)
Comment (brboi) [reply addons/wowl/static/tests/components/dropdown_tests.ts]: In fact, for this test suite, I can simply: 
```suggestion
    parent.unmount();
```

File: addons/wowl/static/tests/components/dropdown_tests.ts (hunk 3 above)
Comment (brboi) [reply addons/wowl/static/tests/components/dropdown_tests.ts:191]: :+1:

File: addons/wowl/static/tests/components/dropdown_tests.ts (hunk 4 above)
Comment (brboi) [reply addons/wowl/static/tests/components/dropdown_tests.ts]: :+1:

File: addons/wowl/static/tests/components/dropdown_tests.ts (hunk 2 above)
Comment (aab-odoo) [reply addons/wowl/static/tests/components/dropdown_tests.ts]: ok then, keep it as it is

Comment (brboi) [reply addons/wowl/static/tests/components/dropdown_tests.ts]: Hmmm finally I just changed it to "startOpen"...
I think this is better than just "open".

"open" would let people think that they may control the opening state from outside.
"startOpen" tells exactly what it does

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**